### PR TITLE
Update checkboxes logic

### PR DESCRIPTION
**Pull request by marianamarasoiu**
With this PR, the conversation list checkboxes will be shown all the time if the `send_multi_message_enabled` flag is set to `true`, as opposed to only when the select all checkbox is ticked. This is to support selecting messages one by one, starting from no selected messages.

### CODE REVIEW

File: webapp/lib/controller.dart [hunk 1]
```diff
@@ -350,7 +348,11 @@ void applyConfiguration(model.UserConfiguration config) {
 
   view.conversationPanelView.showCustomMessageBox(config.sendCustomMessagesEnabled ?? false);
 
-  view.conversationListPanelView.showConversationSelectCheckboxes(config.sendMultiMessageEnabled ?? false);
+  if (config.sendMultiMessageEnabled) {
```
Comment (danrubel) [inline webapp/lib/controller.dart:351]: Would it make more sense to rename this config feature `mulitMessageActionEnabled` or something like that?
Or does it only enable sending multiple messages?

File: webapp/lib/controller.dart [hunk 2]
```diff
@@ -587,19 +586,19 @@ void command(UIAction action, Data data) {
...
-          if (!conversation.unread) {
-            markedConversations.add(conversation);
-            view.conversationListPanelView.markConversationUnread(conversation.docId);
-          }
-        }
-        platform.updateUnread(markedConversations, true).catchError(showAndLogError);
-      } else {
+      if (!currentConfig.sendMultiMessageEnabled || selectedConversations.isEmpty) {
         view.conversationListPanelView.markConversationUnread(activeConversation.docId);
         platform.updateUnread([activeConversation], true).catchError(showAndLogError);
```
Comment (danrubel) [inline webapp/lib/controller.dart:591]: Would it make sense for `selectedConversations` to have the following characteristics?
* always non-null
* always contain `activeConversation` if `activeConversation != null`
* always be empty if `activeConversation == null`

If this were true, then several action handlers like this one could be simplified.

File: webapp/lib/view.dart [hunk 3]
```diff
@@ -917,10 +914,10 @@ class ConversationSummary with LazyListViewItem {
       ..classes.add('conversation-selector')
       ..title = 'Select conversation'
       ..checked = _checked
-      ..style.visibility = 'hidden'
+      ..hidden = _checkboxVisible
```
Comment (danrubel) [inline webapp/lib/view.dart:917]: hidden = visible ?
Based on the code below, perhaps this should be `..hidden = false` ?
And maybe the field initialization above should be `bool _checkboxVisible = true;` to match ?

Comment (marianamarasoiu) [reply webapp/lib/view.dart]: Thanks @danrubel, yeah, this was a bit messy. Switched to a `_checkboxHidden` variable which makes things more straighforward.

File: webapp/lib/controller.dart (hunk 2 above)
Comment (marianamarasoiu) [reply webapp/lib/controller.dart:591]: Thanks @danrubel I'll need to have a think in more detail about this as the separation between them seems helpful - in multiConversationSelect mode, the active conversation is just an indicator for the conversation you're seeing in the second panel which supports a workflow where the user goes through the list, reads each conversation, and only selects it if they want to add it to a bulk reply list. Always considering `activeConversation` as part of the `selectedConversations` would require changing that workflow or doing something else in the UI to support it.

File: webapp/lib/controller.dart (hunk 1 above)
Comment (marianamarasoiu) [reply webapp/lib/controller.dart:351]: yes, I think that would make sense - it also enables tagging multiple conversations and also marking multiple conversations as unread. I'll follow up with a PR to change this. thanks for the suggestion!

**Pull request merged**